modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/ModelRepository.java

```diff
@@ -26,6 +27,10 @@
  * @author kashah
  */
 public interface ModelRepository extends PagingAndSortingRepository<Model, Long> {
+
     List<Model> findByUuid(@Param("uuid") String uuid);
 
+    @Query("select mmm.model from MetricModelMapping mmm where mmm.metric.hash=:hash")
+    List<Model> findModelsByHash(@Param("hash") String hash);
```

Comment:
Let's call this "findByMetricHash". The "findModels" part is implicit since this is on the model repo.